docs/dev/triage-rocmlir.rst

```diff
@@ -0,0 +1,144 @@
+Issue Triaging Guide for suspected rocMLIR issue
+================================================
+
+This document serves as a guide to narrow down whether a bug is due to
+rocMLIR backend of MIGraphX.
+
+There are broadly 3 categories of bugs that can be due to rocMLIR.
+
+1. [B1]rocMLIR compilation bug
```

Comment:
Another debugging check could reverting rocMLIR SHA. 
If there was a known working SHA before try building MIGraphX against that and see if that works without any problems. 
and then try newer rocMLIR SHA and see if it fails or passes with MIGraphX. That could indicate if problem lies in rocMLIR or MIGraphX

Reply:
Yeah

But also, conversely, holding the rocMLIR SHA constant at the newest one while debugging can be useful

Reply:
changing rocMLIR SHA feels like something rocMLIR team should be doing...
I m not sure whether it is advisable to try older SHAs where bugs are fixed in the later ones.

Therefore, I think disabling rocMLIR might be enough with the latest SHA in most cases to see whether its rocMLIR vs MIGraphX bug. (Yes, there can be complex cases where rocMLIR offloads creating a different graph structure exposing a MIGraphX bug but with current infrastructure there is no good way to handle that -- we can have a separate discussion how to improve debuggability) 

I d go further and say unless there is a warning from the following, we should not be changing rocMLIR SHA in debugging.  https://github.com/ROCm/AMDMIGraphX/blob/3e65032266c013e840fd08a2f41623a822e1c538/src/targets/gpu/mlir.cpp#L42-L43